Sources/Foundation/JSONEncoder.swift

```diff
@@ -1152,1359 +1152,801 @@ open class JSONDecoder {
+        func decode(_: Int16.Type) throws -> Int16 {
+            try decodeFixedWidthInteger()
         }
 
-        self.decoder.codingPath.append(key)
-        defer { self.decoder.codingPath.removeLast() }
+        func decode(_: Int32.Type) throws -> Int32 {
+            try decodeFixedWidthInteger()
+        }
 
-        guard let value = try self.decoder.unbox(entry, as: Int32.self) else {
-            throw DecodingError.valueNotFound(type, DecodingError.Context(codingPath: self.decoder.codingPath, debugDescription: "Expected \(type) value but found null instead."))
+        func decode(_: Int64.Type) throws -> Int64 {
+            try decodeFixedWidthInteger()
         }
 
-        return value
-    }
+        func decode(_: UInt.Type) throws -> UInt {
+            try decodeFixedWidthInteger()
+        }
 
-    public func decode(_ type: Int64.Type, forKey key: Key) throws -> Int64 {
-        guard let entry = self.container[key.stringValue] else {
-            throw DecodingError.keyNotFound(key, DecodingError.Context(codingPath: self.decoder.codingPath, debugDescription: "No value associated with key \(_errorDescription(of: key))."))
+        func decode(_: UInt8.Type) throws -> UInt8 {
+            try decodeFixedWidthInteger()
         }
 
-        self.decoder.codingPath.append(key)
-        defer { self.decoder.codingPath.removeLast() }
+        func decode(_: UInt16.Type) throws -> UInt16 {
+            try decodeFixedWidthInteger()
+        }
 
-        guard let value = try self.decoder.unbox(entry, as: Int64.self) else {
-            throw DecodingError.valueNotFound(type, DecodingError.Context(codingPath: self.decoder.codingPath, debugDescription: "Expected \(type) value but found null instead."))
+        func decode(_: UInt32.Type) throws -> UInt32 {
+            try decodeFixedWidthInteger()
         }
-        
-        return value
-    }
-    
-    public func decode(_ type: UInt.Type, forKey key: Key) throws -> UInt {
-        guard let entry = self.container[key.stringValue] else {
-            throw DecodingError.keyNotFound(key, DecodingError.Context(codingPath: self.decoder.codingPath, debugDescription: "No value associated with key \(_errorDescription(of: key))."))
+
+        func decode(_: UInt64.Type) throws -> UInt64 {
+            try decodeFixedWidthInteger()
         }
 
-        self.decoder.codingPath.append(key)
-        defer { self.decoder.codingPath.removeLast() }
+        func decode<T>(_: T.Type) throws -> T where T: Decodable {
+            try self.impl.unwrap(as: T.self)
+        }
 
-        guard let value = try self.decoder.unbox(entry, as: UInt.self) else {
-            throw DecodingError.valueNotFound(type, DecodingError.Context(codingPath: self.decoder.codingPath, debugDescription: "Expected \(type) value but found null instead."))
+        @inline(__always) private func decodeFixedWidthInteger<T: FixedWidthInteger>() throws -> T {
```

Comment:
Isn't the trick for **really really I mean it inline me** `@inlinable @inline(__always)`?

Reply:
`@inlinable` can only be applied to `public` functions AFAIK.

Reply:
Ah true that's the time to use that trick...